ibis/backends/flink/datatypes.py

```diff
@@ -0,0 +1,97 @@
+    @classmethod
+    def from_ibis(cls, schema: sch.Schema | None) -> list[RowType]:
+        if schema is None:
+            return None
+
+        return DataTypes.ROW(
+            [
+                DataTypes.FIELD(k, FlinkType.from_ibis(v))
+                for k, v in schema.fields.items()
+            ]
+        )
+
+
+class FlinkType(TypeMapper):
+    @classmethod
+    def to_ibis(cls, typ: DataType, nullable=True) -> dt.DataType:
+        """Convert a flink type to an ibis type."""
+        if typ == DataTypes.STRING():
+            return dt.String(nullable=nullable)
+        elif typ == DataTypes.BOOLEAN():
+            return dt.Boolean(nullable=nullable)
+        elif typ == DataTypes.BYTES():
+            return dt.Binary(nullable=nullable)
+        elif typ == DataTypes.TINYINT():
+            return dt.Int8(nullable=nullable)
+        elif typ == DataTypes.SMALLINT():
+            return dt.Int16(nullable=nullable)
+        elif typ == DataTypes.INT():
+            return dt.Int32(nullable=nullable)
+        elif typ == DataTypes.BIGINT():
+            return dt.Int64(nullable=nullable)
+        elif typ == DataTypes.FLOAT():
+            return dt.Float32(nullable=nullable)
+        elif typ == DataTypes.DOUBLE():
+            return dt.Float64(nullable=nullable)
+        elif typ == DataTypes.DATE():
+            return dt.Date(nullable=nullable)
+        elif typ == DataTypes.TIME():
+            return dt.Time(nullable=nullable)
+        elif typ == DataTypes.TIMESTAMP():
+            return dt.Timestamp(nullable=nullable)
+        else:
+            return super().to_ibis(typ, nullable=nullable)
+
+    @classmethod
+    def from_ibis(cls, dtype: dt.DataType) -> DataType:
+        """Convert an ibis type to a flink type."""
+        if dtype.is_string():
+            return DataTypes.STRING()
+        elif dtype.is_boolean():
+            return DataTypes.BOOLEAN()
+        elif dtype.is_binary():
+            return DataTypes.BYTES()
+        elif dtype.is_int8():
+            return DataTypes.TINYINT()
+        elif dtype.is_int16():
+            return DataTypes.SMALLINT()
+        elif dtype.is_int32():
+            return DataTypes.INT()
+        elif dtype.is_int64():
+            return DataTypes.BIGINT()
+        elif dtype.is_uint8():
+            return DataTypes.TINYINT()
+        elif dtype.is_uint16():
+            return DataTypes.SMALLINT()
+        elif dtype.is_uint32():
+            return DataTypes.INT()
+        elif dtype.is_uint64():
+            return DataTypes.BIGINT()
+        elif dtype.is_float16():
+            return DataTypes.FLOAT()
+        elif dtype.is_float32():
+            return DataTypes.FLOAT()
+        elif dtype.is_float64():
+            return DataTypes.DOUBLE()
+        elif dtype.is_date():
+            return DataTypes.DATE()
+        elif dtype.is_time():
+            return DataTypes.TIME()
+        elif dtype.is_timestamp():
+            return DataTypes.TIMESTAMP()
```

Comment:
Is there any nullability information we need to pass on to Flink?

Also, definitely not a blocker, but if you have a dictionary mapping the ibis types to the flink types it can make it easier to quickly glance and see how the types are mapped between the two

Reply:
good catch, fixed the nullable types